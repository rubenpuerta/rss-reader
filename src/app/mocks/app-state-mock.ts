import { NewsItem } from '@models/rss-news.model';

export const mockedNews: NewsItem[] = [
	{
		author: 'test-name',
		description: 'test-html-code',
		guid: 'test-guid',
		pubDate: 12345,
		title: 'test-title',
		imageUrl: 'test-url-path-image',
		articleBrief: 'test-article-brief'
	}
];

export const defaultAppState: any = {
	commonUI: {
		isLoading: false
	},
	rssItems: {
		news: mockedNews
	}
};

export const getMockState = (stateChanges?: Partial<any>): any =>
	({
		...defaultAppState,
		...stateChanges
	} as any);
