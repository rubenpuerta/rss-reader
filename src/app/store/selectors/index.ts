export * from './rss.selectors';
export * from './ui.selectors';
