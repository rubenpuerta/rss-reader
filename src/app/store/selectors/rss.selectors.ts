import { createFeatureSelector, createSelector } from '@ngrx/store';

import { RssState } from '@store/reducers/rss.reducer';

const getRssItemsState = createFeatureSelector<RssState>('rssItems');

export const getRssFeed = createSelector(getRssItemsState, ({ news }) => news);
