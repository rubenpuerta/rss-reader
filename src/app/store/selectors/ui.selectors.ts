import { createFeatureSelector, createSelector } from '@ngrx/store';

import { CommonUIState } from '@store/reducers';

const getCommonUIState = createFeatureSelector<CommonUIState>('commonUI');

export const getIsAppLoading = createSelector(getCommonUIState, (uiState) => uiState?.isLoading);
