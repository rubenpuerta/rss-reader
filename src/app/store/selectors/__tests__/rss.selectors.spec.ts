import { mockedNews } from '@mocks/app-state-mock';
import { rssInitialState, RssState } from '@store/reducers/rss.reducer';
import { getRssFeed } from '@store/selectors';

const getRssState = (newValue?: Partial<RssState>) => ({
	rssItems: {
		...rssInitialState,
		...newValue
	}
});

describe('RssSelectors', () => {
	it('should return Rssstate', () => {
		const result = getRssState();
		expect(result).toMatchSnapshot();
	});

	it('should return an empty array', () => {
		const result = getRssFeed(getRssState());
		expect(result).toMatchSnapshot();
	});

	it('should return an array with Rss', () => {
		const store = getRssState({ news: mockedNews });
		const result = getRssFeed(store);
		expect(result).toMatchSnapshot();
	});
});
