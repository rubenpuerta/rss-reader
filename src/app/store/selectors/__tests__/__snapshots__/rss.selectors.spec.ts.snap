// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RssSelectors should return Rssstate 1`] = `
Object {
  "rssItems": Object {
    "news": Array [],
  },
}
`;

exports[`RssSelectors should return an array with Rss 1`] = `
Array [
  Object {
    "articleBrief": "test-article-brief",
    "author": "test-name",
    "description": "test-html-code",
    "guid": "test-guid",
    "imageUrl": "test-url-path-image",
    "pubDate": 12345,
    "title": "test-title",
  },
]
`;

exports[`RssSelectors should return an empty array 1`] = `Array []`;
